### ISSUE TITLE
Dropout still active in .eval() mode

### ISSUE THREAD

**Issue by denjots** [pytorch/pytorch#11999]
Using Pytorch 1.0 preview (nightly build) I'm finding that dropout is still left enabled during testing even though .eval() is used.

**Comment by crcrpar** [comment]
Could you share a minimal code to reproduce that?

In my envrionment, dropoout worked correctly as below.
```python
In [1]: import torch, torch.nn as nn, torch.nn.functional as F

In [2]: x = torch.randn(3, 3, dtype=torch.float)

In [3]: dropout = nn.Dropout()

In [4]: dropout(x)
Out[4]: 
tensor([[-0.0000, -0.0000,  3.6885],
        [ 0.0000,  0.6140, -0.0000],
        [-0.9087, -0.0000,  0.2604]])

In [5]: dropout.eval(); dropout(x)
Out[5]: 
tensor([[-2.2393, -0.2917,  1.8443],
        [ 1.2517,  0.3070, -0.8705],
        [-0.4543, -1.4322,  0.1302]])

In [6]: x
Out[6]: 
tensor([[-2.2393, -0.2917,  1.8443],
        [ 1.2517,  0.3070, -0.8705],
        [-0.4543, -1.4322,  0.1302]])

In [7]: dropout.train(); dropout(x) / x
Out[7]: 
tensor([[2., 0., 2.],
        [2., 2., 2.],
        [0., 0., 0.]])

In [8]: dropout.eval(); dropout(x) / x
Out[8]: 
tensor([[1., 1., 1.],
        [1., 1., 1.],
        [1., 1., 1.]])

In [9]: torch.__version__
Out[9]: '1.0.0.dev20180921'
```

**Comment by denjots** [comment]
More testing suggest that it's the use of the functional API version of dropout in the forward section of my NN model definition that's the problem. In 0.4.1 the two print statements output the same number (no dropout) - in the 1.0.0 preview different numbers (dropout active). See below...

```
import torch
import torch.nn as nn
import torch.nn.functional as F

class ConvNet(nn.Module):
    def forward(self, x):
        out = F.dropout(x, p=0.5)
        return out

network = ConvNet()

network.eval()

with torch.no_grad():
    inputs = torch.ones(1, 64, 100, 100, dtype=torch.float32)
    output = network(inputs)
    print(torch.sum(output).item())
    output = network(inputs)
    print(torch.sum(output).item())
```
